.github/workflows/examples.yml

```diff
@@ -21,51 +21,51 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        go-version:
+        goversion:
           - "1.20.x"
           - "1.21.x"
     runs-on: ubuntu-latest
     timeout-minutes: 30
     steps:
+      - uses: actions/checkout@v4
```

Comment:
moving the actions/checkout before setup-go, because setup-go is trying to cache dependencies based on go.mod and go.sum